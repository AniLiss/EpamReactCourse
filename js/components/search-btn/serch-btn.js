import React from 'react';

export const SearchButton = () => {
    return (
        <button className="search-btn">Search</button>
    )
};
