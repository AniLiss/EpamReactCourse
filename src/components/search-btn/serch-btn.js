import React from 'react';
import './search-btn.scss'
export const Button = ({text, active, light, onClick, dataId}) => {
    const handleClick = (...args) => {
            onClick(...args);
    };

    const isActive = active ? "search-btn--active"  : '';
    const isLight = light ? "search-btn--light"  : '';
    const classNames = `search-btn ${isActive} ${isLight}`;

    return (
        <button data-id={dataId} onClick={handleClick} className={classNames}>{text}</button>
    )
};
