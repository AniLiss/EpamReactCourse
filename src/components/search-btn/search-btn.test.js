import React from 'react';
import {mount, shallow} from 'enzyme';
import { Button } from './serch-btn';

describe('Button', () => {
    it('is expected to render default Button correctly', () => {
        const wrapper = shallow(<Button text="default btn" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('is expected to render active Button correctly', () => {
        const wrapper = shallow(<Button active text="active btn"/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('is expected to render light Button correctly', () => {
        const wrapper = shallow(<Button light text="light btn"/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('is expected to handle Button click', () => {
        const onClick = jest.fn();
        const wrapper = mount(
            <Button onClick={onClick} />
        );
        wrapper.find('.search-btn').at(0).simulate('click');
        expect(onClick).toHaveBeenCalled();
    });
});
