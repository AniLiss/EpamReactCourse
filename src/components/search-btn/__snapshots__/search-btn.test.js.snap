// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is expected to render active Button correctly 1`] = `
<button
  className="search-btn search-btn--active "
  onClick={[Function]}
>
  active btn
</button>
`;

exports[`Button is expected to render default Button correctly 1`] = `
<button
  className="search-btn  "
  onClick={[Function]}
>
  default btn
</button>
`;

exports[`Button is expected to render light Button correctly 1`] = `
<button
  className="search-btn  search-btn--light"
  onClick={[Function]}
>
  light btn
</button>
`;
